### PR TITLE
WIP: Resolve "Real Documentation"

### LINKED ISSUE #85: Make a real documentation
Now that DiffEqFlux, ModelingToolkit, Surrogates, etc. all have one, NeuralNetDiffEq needs to get one.

Remaining documentation changes:

- [ ] Split the tutorials into separate ones on the sidebar. Have "copy-pastable code" on top, explained code below.
- [ ] Tutorials for the neural network ODE solver by using the tests
- [ ] More explanation in the deep BSDE methods as to where they come from and what they solve
- [ ] Separate out reference documentation
- [ ] For the reference documentation, convert some of the stuff from the README into docstrings, and use the docstrings in the documentation

https://github.com/SciML/NeuralNetDiffEq.jl/pull/86 for more details. I think that's what's needed to get it released, and then we can keep going from there!

### PR DESCRIPTION
**Pull request by zzj0402**
- [ ] ## Split the tutorials into separate ones on the sidebar. Have "copy-pastable code" on top, explained code below.
- [x] ### Split 100hjb example
- [x] ### ODE
- [ ] ### Kolmogorov Equations
- [x] ### High Dimensional PDEs
- [x]  ## Tutorials for the neural network ODE solver by using the tests
- [ ]  ## More explanation in the deep BSDE methods as to where they come from and what they solve
- [ ]  ## Separate out reference documentation
- [ ]  ## For the reference documentation, convert some of the stuff from the README into docstrings, and use the docstrings in the documentation

closes #85

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "cSpell.words": [
```
Comment (ChrisRackauckas) [inline .vscode/settings.json:2]: this seems misplaced?

File: README.md [hunk 2]
```diff
@@ -7,15 +7,21 @@
 [![codecov.io](http://codecov.io/github/JuliaDiffEq/NeuralNetDiffEq.jl/coverage.svg?branch=master)](http://codecov.io/github/JuliaDiffEq/NeuralNetDiffEq.jl?branch=master)
 
 The repository is for the development of neural network solvers of differential equations such as physics-informed
-neural networks (PINNs) and deep BSDE solvers. It utilizes techniques like deep neural networks and neural 
+neural networks (PINNs) and deep BSDE solvers. It utilizes techniques like deep neural networks and neural
 stochastic differential equations to make it practical to solve high dimensional PDEs efficiently through the
 likes of scientific machine learning (SciML).
```
Comment (ChrisRackauckas) [inline README.md:12]: The README should probably be mostly deleted, like DiffEqFlux, and just pass people to the docs.

File: docs/src/index.md [hunk 3]
```diff
@@ -1,124 +1,53 @@
 # NeuralNetDiffEq
 
-The repository is for the development of neural network solvers of differential equations such as physics-informed
-neural networks (PINNs) and deep BSDE solvers. It utilizes techniques like deep neural networks and neural 
-stochastic differential equations to make it practical to solve high dimensional PDEs efficiently through the
```
Comment (ChrisRackauckas) [inline docs/src/index.md:5]: There are a lot of grammatical errors in the edits made to this file.

File: docs/src/index.md [hunk 4]
```diff
@@ -1,124 +1,53 @@
...
-likes of scientific machine learning (SciML).
+NeuralNetDiffEq.jl consists neural network solvers for differential equations, such as physics-informed neural networks (PINNs), and deep BSDE solvers.
```
Comment (ChrisRackauckas) [inline docs/src/index.md:3]: consists of neural network solvers for differential equations such as ...

File: docs/src/index.md [hunk 5]
```diff
@@ -1,124 +1,53 @@
...
-tspan = (0.0f0, 1.0f0)
-λ = 1.0f0
-
-g(X) = log(0.5f0 + 0.5f0*sum(X.^2))
-f(X,u,σᵀ∇u,p,t) = -λ*sum(σᵀ∇u.^2)
-μ_f(X,p,t) = zero(X)  #Vector d x 1 λ
-σ_f(X,p,t) = Diagonal(sqrt(2.0f0)*ones(Float32,d)) #Matrix d x d
-prob = TerminalPDEProblem(g, f, μ_f, σ_f, X0, tspan)
-```
+Considering S be a solution process to the SDE:
```
Comment (ChrisRackauckas) [inline docs/src/index.md:21]: Consider S as the solution process of the SDE:

etc. there's more like this, so these edits probably need a second read

Comment (zzj0402) [reply docs/src/index.md:21]: I just moved old readme around. I will just rewrite them then.

File: .vscode/settings.json (hunk 1 above)
Comment (zzj0402) [reply .vscode/settings.json:2]: It's for vscode spell checker. I can add this to ignore file.

Comment (ChrisRackauckas) [reply .vscode/settings.json:2]: yeah I'm not sure this should go into the repo

**Comment by zzj0402** [comment]
## UAT
## Prerequisites
### Julia
### IDE
### CUDA
### Packages

**Comment by zzj0402** [comment]
- [ ] Solving Differential Equations with DifferentialEquations.jl
- [ ] Solving ODEs with Neural Networks
- [ ] Coding Up the Method

**Comment by ashutosh-b-b** [comment]
Hi @zzj0402 Can u share with me the progress of the documentation so that we can divide it without overlapping?

**Comment by zzj0402** [comment]
All above are the progress

**Comment by ChrisRackauckas** [comment]
> Solving Differential Equations with DifferentialEquations.jl

That's not necessary for this documentation.

> Coding Up the Method

That probably is unnecessary here too, since the documentation is about using the method not about how it's written.

**Comment by zzj0402** [comment]
Sorry, I was using that for learning tracking.

**Comment by zzj0402** [comment]
> Hi @zzj0402 Can u share with me the progress of the documentation so that we can divide it without overlapping?

You may work from bottom-up in the todo list. We will meet in the middle.

**Comment by zzj0402** [comment]
https://stackoverflow.com/questions/62747641/whats-the-neuralnetdiffeq-jl-equivalent-of-this-piece-of-code-for-solving-ode-i

**Comment by ashutosh-b-b** [comment]
> https://stackoverflow.com/questions/62747641/whats-the-neuralnetdiffeq-jl-equivalent-of-this-piece-of-code-for-solving-ode-i

Hi! for `du/dt = f(u , p, t)` we first define a `f` and then define an `ODEProblem` passing `f` , `u0` and `tspan`. For alg parameter we pass `NeuralNetDiffEq.NNODE(chain,opt)` . You can look into the test cases to see applications.

**Comment by zzj0402** [comment]
https://www.pnas.org/content/115/34/8505

**Comment by zzj0402** [comment]
https://www.mathematik.hu-berlin.de/~perkowsk/files/bsde.pdf

File: .gitignore [hunk 6]
```diff
@@ -2,3 +2,5 @@
 *.jl.*.cov
 *.jl.mem
 Manifest.toml
+*.pdf
+.vscode/settings.json
```
Comment (ChrisRackauckas) [inline .gitignore:6]:
```suggestion
```

**Comment by ChrisRackauckas** [comment]
Note that I didn't want this to keep blocking our other developers, so I went ahead and fixed this up and merged. @KirillZubov @ashutosh-b-b @zzj0402 all should please check out the documentation and should start hammering away at it. There a lot that is done, but still many todos and as always it can be better. @kanav99 it would be great if you could try and setup the deployment?

**Comment by ChrisRackauckas** [comment]
Note that the README will be left completely intact until the documentation is deployed, then we'll just pair it down like the ModelingToolkit.jl repo, i.e. a quick explanation that sends people to the docs, but some pretty pictures to make them happy. For that, we can have the 2-dimensional Poisson equation example with pretty pictures, and then the 100-dimensional HJB, with no extra explanations but just to drive people to the docs. So @KirillZubov try and prepare a PR for this ASAP.

**Pull request closed without merging**